### DISCUSSION TITLE
How to set OGR_XLSX_HEADERS

### DISCUSSION THREAD

**Discussion by imadhajjchahine** [duckdb/duckdb-spatial#97] [Q&A] [answered]
Hi,

In duckdb CLI I have installed and load the spatial extension, and I am trying to 
SET OGR_XLSX_HEADERS='FORCE';
Error: Catalog Error: unrecognized configuration parameter "OGR_XLSX_HEADERS"

Any guide on how to set this flag and how to set on per query basis.

Thanks

**Comment by Maxxen** [comment, marked as answer]
EDIT:
Implemented in #114 

Hi!

It is not possible to set this on a per-query basis as of now. You should be able to set it as a environment variable so GDAL can pick it up. https://gdal.org/user/configoptions.html#configoptions

Im currently on vacation, but In the future ill look into making it possible to set these sorts of options directly from duckdb.

**Reply by davidjbrennan** [reply]
It would be so useful if these options could be set as "open options" rather than the current "configuration options" so that every query didn't have to use the same options.  

Hope you are able to look in to this and make it possible.

**Reply by Maxxen** [reply]
Thanks for the feedback, I've implemented a fix for this in #114

**Comment by Maxxen** [comment]
Implemented in #114

**Comment by Maxxen** [comment]
Hello!

As of DuckDB 1.2.0 the duckdb [excel extension](https://duckdb.org/docs/extensions/excel.html#reading-xlsx-files) provides support for reading and writing xlsx files with greater efficiency and many more options. Therefore we're going to deprecate this feature in the spatial extension in the future. I would recommend that you try out the excel extension, and if you run into any problems please open an issue over at the [duckdb-excel repository](https://github.com/duckdb/duckdb-excel).
